tests/GitWorkflowTest.php

```diff
@@ -504,4 +504,70 @@ public function executeCommand(string $command, array &$output = null, int &$ret
 		$messages = runGitWorkflow([$gitFile], $options, $shell, $debug);
 		$this->assertEquals($expected->getMessages(), $messages->getMessages());
 	}
+	function testFullGitWorkflowForInterBranchDiff() {
+		$gitFile = 'bin/foobar.php';
+		$debug = function($message) { var_dump( $message ); }; //phpcs:ignore VariableAnalysis
+		$shell = new class() implements ShellOperator {
+			public function isReadable(string $fileName): bool {
+				return ($fileName === 'bin/foobar.php');
+			}
+
+			public function exitWithCode(int $code): void {} // phpcs:ignore VariableAnalysis
+
+			public function printError(string $message): void {} // phpcs:ignore VariableAnalysis
+
+			public function validateExecutableExists(string $name, string $command): void {} // phpcs:ignore VariableAnalysis
+
+			public function executeCommand(string $command, array &$output = null, int &$return_val = null): string {
+				var_dump( $command );
+				if (false !== strpos($command, "git diff 'master'... --no-prefix 'bin/foobar.php'")) {
+					$return_val = 0;
+					return <<<EOF
+diff --git bin/foobar.php bin/foobar.php
+index c012707..319ecf3 100644
+--- bin/foobar.php
++++ bin/foobar.php
+@@ -3,6 +3,7 @@
+ use Billing\Purchases\Order;
+ use Billing\Services;
+ use Billing\Ebanx;
++use Foobar;
+ use Billing\Emergent;
+ use Billing\Monetary_Amount;
+ use Stripe\Error;
+EOF;
+				}
+				if (false !== strpos($command, "git status --short 'bin/foobar.php'")) {
+					$return_val = 0;
+					return ''; // the file is not modified, as this runs on fresh checkout.
+				}
+				if ( false !== strpos($command, "git cat-file -e 'master':'bin/foobar.php'")) {
+					$return_val = 0;
+					return '';
+				}
+				if ( false !== strpos($command, "git show 'master':$(git ls-files --full-name 'bin/foobar.php') | phpcs --report=json -q --stdin-path='bin/foobar.php' -") ){
+					return '{"totals":{"errors":0,"warnings":1,"fixable":0},"files":{"\/srv\/www\/wordpress-default\/public_html\/test\/bin\/foobar.php":{"errors":0,"warnings":1,"messages":[{"message":"Found unused symbol '."'Billing\\\\Emergent'".'","source":"ImportDetection.Imports.RequireImports.Import","severity":5,"fixable":false,"type":"WARNING","line":6,"column":5}]}}}';
+				}
+				if ( false !== strpos($command, "cat 'bin/foobar.php' | phpcs --report=json -q --stdin-path='bin/foobar.php' -")) {
+					return '{"totals":{"errors":0,"warnings":2,"fixable":0},"files":{"\/srv\/www\/wordpress-default\/public_html\/test\/bin\/foobar.php":{"errors":0,"warnings":2,"messages":[{"message":"Found unused symbol '."'Foobar'".'.","source":"ImportDetection.Imports.RequireImports.Import","severity":5,"fixable":false,"type":"WARNING","line":6,"column":5},{"message":"Found unused symbol '."'Billing\\\\Emergent'".'.","source":"ImportDetection.Imports.RequireImports.Import","severity":5,"fixable":false,"type":"WARNING","line":7,"column":5}]}}}';
+				}
+				
+				return '';
```

Comment:
Can this be replaced with `throw new \Exception("Unknown command: {$command}");` like the other tests? That way the test makes sure that there are no un-mocked commands trying to be run that might skew the test results in the future.

Reply:
Looks like I have started off a copy one of the few class implementations in the file which does not throw the exception :) Fixed.